packages/flutter_tools/templates/plugin/.idea/libraries/Dart_SDK.xml.tmpl

```diff
@@ -0,0 +1,25 @@
+<component name="libraryTable">
+  <library name="Dart SDK">
+    <CLASSES>
+      <root url="file://{{dartSdk}}/lib/async" />
+      <root url="file://{{dartSdk}}/lib/collection" />
+      <root url="file://{{dartSdk}}/lib/convert" />
+      <root url="file://{{dartSdk}}/lib/core" />
+      <root url="file://{{dartSdk}}/lib/developer" />
+      <root url="file://{{dartSdk}}/lib/html" />
+      <root url="file://{{dartSdk}}/lib/indexed_db" />
+      <root url="file://{{dartSdk}}/lib/io" />
+      <root url="file://{{dartSdk}}/lib/isolate" />
+      <root url="file://{{dartSdk}}/lib/js" />
+      <root url="file://{{dartSdk}}/lib/js_util" />
+      <root url="file://{{dartSdk}}/lib/math" />
+      <root url="file://{{dartSdk}}/lib/mirrors" />
+      <root url="file://{{dartSdk}}/lib/svg" />
+      <root url="file://{{dartSdk}}/lib/typed_data" />
+      <root url="file://{{dartSdk}}/lib/web_audio" />
+      <root url="file://{{dartSdk}}/lib/web_gl" />
+      <root url="file://{{dartSdk}}/lib/web_sql" />
+    </CLASSES>
```

Comment:
It can't be in the flutter repo because it's how the IDE locates the flutter repo. (Also, everything in the .idea directory is part of IntelliJ's world so we can't just change the file location or format.)

An alternate approach would be to record the position of the Flutter SDK somewhere else, and the Flutter plugin can set the location of the Dart SDK on project open, which will cause the Dart plugin to generate this file. That way it doesn't have to be in the template.

So far the best idea seems to be to derive it from the .packages file. That would work fine for `flutter create` but maybe not so great after doing a `git clone` on a Flutter app's repo, since "flutter packages get" hasn't been run yet. Then again, the location of the flutter repo is machine-specific, so making the user choose it might not be so bad?

(I think @pq has some plans to work on the getting started experience.)